### DISCUSSION TITLE
Update the window's contents?

### DISCUSSION THREAD

**Discussion by tseli0s** [rust-windowing/winit#2407] [Q&A] [answered]
# Update the window's contents?
So when I create the window, how exactly do I update it's contents?

I have seen things being mentioned like `window.request_redraw()` and `winit::event::Event::RedrawRequested`, but I'm a bit confused on when the latter is received (I know the former one emits that event), and if it's always received or just on resizes. I kind of think this way because I've worked with GLFW and SDL in C/C++ where the updating is done through buffer swapping, and never really cared about how a window works in the first place.

**Comment by kchibisov** [comment, marked as answer]
winit doesn't draw and it doesn't provide calls to the graphics platform api, unlike glfw/sdl. The `RedrawRequested` event is simply to tell you that you should draw, e.g. due to resize or some other system event. How you'd draw is entirely up to you. You can use OpenGL (with glutin) or Vulkan(with ash) and call their routines to present the buffer.

**Reply by tseli0s** [reply]
Oh, ok :) I thought request_redraw() meant updating the window, thanks for the information !
